### ISSUE TITLE
Snippets in assists don't work anymore

### ISSUE THREAD

**Issue by jonas-schievink** [rust-analyzer/rust-analyzer#8208]
We have a custom LSP extension for snippets in assists, it doesn't seem to work anymore:

```rust
fn f() {
    $0()$0
}
```

Apply "Extract variable"

```rust
fn f() {
    let $0var_name = ();
    var_name
}
```

...but there's no snippet

**Comment by ivan770** [comment]
Is this reproducible editors other than VS Code? Trace logs show that we send snippet insert text format

**Comment by ivan770** [comment]
Got it narrowed down to `placeholderLength` always being at zero, thus putting cursor at the beginning of placeholder but without selecting it

**Comment by jonas-schievink** [comment]
It seems that this was never supported like I thought it was, but I've fixed some bugs around this in https://github.com/rust-analyzer/rust-analyzer/pull/8210

**Comment by jonas-schievink** [comment]
Closing, since the known issues were fixed in #8210, even though that didn't end up using this feature
